runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -5928,6 +5928,78 @@ class CompilationDensity
       }
    }; // class CompilationDensity
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+static void suspendSamplerThreadForCheckpoint(J9VMThread *samplerThread, J9JITConfig *jitConfig, TR::CompilationInfo *compInfo)
+   {
+   compInfo->acquireCompMonitor(samplerThread);
```

Comment:
This monitor acquire is going to be done every 10 ms. Can we avoid it by checking a flag first and only then do the proper synchronization?